### ISSUE TITLE
Ethereum Classic Treasury Roadmap Call: Initial Discussions

### ISSUE THREAD

**Issue by YazzyYaz** [ethereumclassic/ECIPs#332]
# ETC Core Devs Call - Treasury Roadmap Discussions
When: Thursday, August 13, 2020, 3pm UTC, 60 minutes max.

Where: Ethereum Classic Discord https://discord.gg/dwxb6nf #ecips channel. Will use/create a voice channel ad hoc.

Voice Channel Used will be: `#Treasury-Call-2020-08-13`
Chat Channel on Discord to ask Questions is `#treasury-call-2020-08-13`

**Note**: This is a call only about the treasury roadmap. Any proposals for dealing with 51% attacks have to be on a separate call. See [here](https://github.com/ethereumclassic/ECIPs/issues/333)

## Agenda
*  Quick client teams check-in 
        * Core-Geth - ETC Labs
        * Mantis - IOHK
        * Hyperledger Besu - ETC Coop/ChainSafe
* Introduction for this call
* IOHK Proposal for an Ethereum Classic Treasury
* Gather Community feedback on Proposal via Q&A
* Measure Rough Consensus to Determine if IOHK will move forward with making an ECIP for the ETC Treasury.

**Comment by YazzyYaz** [comment]
Note: this is a call only about the treasury roadmap. Any proposals for dealing with 51% attacks have to be on a separate call.
Any questions related to the treasury can be asked here and I will ask them in the call if needed.

**Comment by pezzullig** [comment]
Really hope you go for IOHK, especially after watching Charles proposed solution video. Could be a fantastic roadmap for ETC. 🙏

**Comment by gitr0n1n** [comment]
ref: https://iohk.io/en/research/library/papers/a-proposal-for-an-ethereum-classic-treasury-system/
ref: https://github.com/ethereumclassic/ECIPs/pull/229
ref: https://youtu.be/oHUQuXOwYeU
ref: https://theconversation.com/a-brief-history-of-gnupg-vital-to-online-security-but-free-and-underfunded-80800
ref: https://youtu.be/Cspqt-nZqsc
ref: https://github.com/ethereumclassic/ECIPs/issues/332

**Comment by Dexaran** [comment]
@bobsummerwill expressed that "there is broad consensus that changing the block rewards to add treasury model is unacceptable" here https://github.com/ethereumclassic/ECIPs/pull/229

What event changed this consensus and at what point in time so that we are discussing an `already rejected dozen of times proposal` again?  Personally, I would like to hear Bob Summewill's opinion on his own position.

**Comment by Dexaran** [comment]
Also it is worth to reference my Security Auditing DAO proposal that was rejected because it required Treasury: https://github.com/ethereumclassic/ECIPs/pull/231

I would like to correct @bobsummerwill and @meowsbits comments.

>Auditing is a good idea, but should not be centralized into the block rewards.

>Agree to Reject on grounds of centralized system services. Don't want to spend any time on it at any meeting.

The proposed auditing system is intended to be a DAO governed by a smart-contract. The proposed system must allow security auditors to enter, leave, pick tasks and get paid based on the quality of work done at any time. 

This is not centralized as it is not tied to any trusted authority that can not be removed. Not really more centralized than [McAfeeDEX](https://mcafeedex.com/#/erc20) that is a smart-contract or [ENS](https://docs.ens.domains/) that also relies on a smart-contract.

In presence of the Treasury system proposal I would like to notice that I'm still capable of building such a decentralized security auditing core system in any project that has built-in governance/funding mechanic.

**Comment by gitr0n1n** [comment]
"Nothing wrong with VC money, but if its the sole way to support development, they will be calling all shots. I really suggest people in this room take a look at the Monero funding mechs and how they support a full time cryptography PhD. There is lot too learn there, just as we are learning from ETH with solutions like MolachDAO.

Personally my belief is the more avenues for developers to feed their families and contribute to this truly decentralized EVM, the better. VC + DAOs + public good useage revenue generators (uniswap, defi, liquidity pools) + product/services kicking in (blockstream or grayscale approach) + bounties (gitcoin) + community pooling (monero) + miners kicking in (voluntarily) + many more approaches. I'm open for anything that pushes for feeding developers and facilitating  a postive feedback loop that will appreciate ETC value/utility." - my post from May in the `community fund` channel.

I think https://github.com/ethereumclassic/ECIPs/pull/231 highlights centralization in development.

**Comment by gitr0n1n** [comment]
"Rough idea: We can build funding sources by porting over high value dapps like uniswap and including a development fee/tax in those. These are public goods that can feed useage funds to the DAO. Then the DAO solicits to build another public good that has a similar development fee built in. A positive feedback loop.

Effectively the public good dapps increase the use cases for the network appreciating ETC's utility. Which increases the transaction counts per block which is more revenue for the miners. And if the public good is generating funding for development, the DAO will now be able to feed developers. So now a long term sustainable model starts forming." - my post from May in the community fund channel.

**Comment by gitr0n1n** [comment]
"This project is SEVERELY under funded and there does not appear to be any decentralized effort to fund it." - my post from January in the `CommunityFund` channel.
"it's kind of lunacy to prefer backroom meetings for funding the open source project, rather than an extremely transparent, public way imo." - my post from January in the `CommunityFund` channel.

**Comment by q9f** [comment]
I would emphasize that this call is for treasury only and created two other calls for technical discussion, especially mining algorithm and other protective measures: #333

**Comment by w1g0** [comment]
The idea of a treasury system sounds very appealing, tempting even for others. Once Charles from IOHK joined the conversation the hopes of everyone rose sky high and I welcome that warmly. The drive in his speech is motivating and what he says about innovating and diversification is key towards the way out of this blind alley. Unfortunately that's where the euphoria ends, because the treasury system goes against all principles where open source community stands for. Money solves the solution in your head, projected into the future. Once reality kicks in you will see that money will create more problems than it solves. People who don't have any money will say I'm wrong, they who have money will understand what I'm talking about.
Anyways then there's the price and content. As of now I haven't heard a price for the product, implementation, nor for continuous support. Remember that Charles mentioned at least ten times he spent 1.5 million on the solution, he also left this decentralized project to start a governed centralized project of its own and that spirit is rooted in all of his proposals. There's no denying that he likes money and control.
The fact there's a miners reward is that they have to invest (meaning risking a return in the future) in physical hardware and pay the bills to keep it running. Yes, developers also need a computer and that also needs juice but doesn't come near the amount invested by miners. I'm not saying here that developers must work for free and they cannot accept any money for the important and sometimes brilliant work they deliver. But saying we try a funding campaign and it didn't work is like saying I tried to wash the car but my arm isn't long enough to get around, try harder! Get creative, just like the innovation that is needed to boost the possibilities for the $etc road map. You will see once tipping becomes effortless and you create value there will be individuals or even groups that will fund your project. Only disbelief can block that reality and if you deliver decent work than the reward will be greater that the meager fund you would've received from the treasury system but therefore you have to earn it.
There's also the need for creating a community with more respect towards each other, encouragement, creating a vision and designing a clear path towards that goal without losing the principles that makes the heart of Ethereum Classic pump.
Let the Phoenix spread its wings and rise from the ashes! <3

**Comment by meowsbits** [comment]
In review of the document at [A Proposal for An Ethereum Classic Treasury System](https://iohk.io/en/research/library/papers/a-proposal-for-an-ethereum-classic-treasury-system/)...

- _Voters_ are defined as "stakers" willing to freeze >=500 ETC (S2.4.1) as collateral against the right to vote, and are incentivized to do so with a "voter reward" calculated at the end of the epoch. A voter is allowed to cast a vote in the opinions of Accept, Abstain, or Reject. This means that all stakeholders who are willing to hold assets for or beyond each `3*epoch`  are incentivized to participate _at least_ in opinion-abstinence (Abstain). So stakeholders can garner rewards (proportional to their stake) without actually contributing to the democratic process (neither For nor Against proposals). This seems like a significant inefficiency, suggesting perhaps that the Treasury as proposed is in effect a bank, whose interest rates are a.) funded by the Miners, b.) access-controlled by miners (availability), and c.) proportional to capital stake. Removing the _Abstain_ vote would promote no-op or malicious proposals, or? See also S4.5.

- Since _Voter Reward_ is a function of _Voter Participation_ (whether via an on-chain smart contract or "special" new transaction types), could this imply an incentive for an miner(s) to censor voter participation by excluding voting transactions (except for their own, of course) from blocks? Unfortunately, I don't see this type of DoS attack this covered in S5.2. 

I'm not sure I see an alignment of the interests (and necessary mechanisms) of democracy and the interests of stakeholder rewards. And if they are not aligned, then what keeps miners from staking only their own votes to garner disproportional and undemocratic marginal rewards on their own stake?... "indirect" supposed externalities, like the price of ETC falling because the Treasury wasn't working?

Have I got something wrong here?

**Comment by q9f** [comment]
My protocol. I'm sorry for missing some questions and answers, if you have anything to fill in the gaps, please comment below.

---

attendees (83):
- al
- alanmcsherry
- alisonalexis
- andrew t.
- austin
- beautybubble
- belfordz
- bobsummerwill
- bpmckenna
- brady990
- buzzlite
- bwana
- carlosm
- classic kev co-op
- cyb3rn00b
- cyrille
- dan_universalcoin
- dolphin134
- donsyang
- dontpanicburns(chipprbots)
- draxpat
- dreamer
- dundidit
- econymous
- eddieoz
- enthef
- erra
- ey
- felipe duarte
- fiftyonefifty
- harishv
- hitchcott
- hudson jameson (souptacular)
- hvetc
- iogcharles
- ipstop
- isaac
- jameswo
- joe stack
- joseph ferrone
- karthik
- kimi sian-yu chen
- klex22
- kru
- liz
- mahof
- marblemadness
- mareliki
- masterdubs | pril founder
- mcdappas
- memur01
- miko
- mk
- mlagattuta
- mr robot
- nautical_illusions
- omniedge
- phyro
- q9f
- quietnan
- rob (canarigs mining)
- rom
- scosol
- sebastien
- sncs
- sonnyrollins
- stev
- superfreek
- tj
- tms88
- tomt
- tradybtc
- vikpat
- w1g0
- winding-coils
- wracul
- x86ik
- xocel
- yaz
- yolobear
- zcstarr
- zois
- dirtyrusski

teams check in:
- core geth: isaac, et al.
- besu: yaz, bob, et al.
- iohk: charles, et al.

intro by yaz:
- charles comes up with treasury idea for etc
- treasury idea as old as ethereum classic
- yaz also wrote a treasury proposal a while back

proposal by charles:
- happy to be back on discord
- just as back in the good old eve online days
- proposal for a treasury hardfork
  - block reward managed by contract
  - contract managed by the entities contributing key infrastructure ("3" entities)
  - easy contract, easy to implement, calling it "proto treasury"
  - block rewards can go to miners, stakeholders, infrastructure providers, ...
  - ... (missing some details here)
- vitalik never posted on the cardano subreddit until the treasury debate

q&a:
- isaac asks about design specifics
	- charles consideres ballots and different voting mechanisms
- chris: you mentioned that a supermajority of etc would be able to shut it off. do we even know that 70%+ of etc is still active? also, how will you ensure that the 'simple' phase one moves to phase 2, as there is a disincentive for 3 core teams to do this. 
  - charles answers this one (missed some bits)
- who are the 3-4 teams that will be funded and how are they decided?
  - charles: the people maintaining core clients: there are three - mantis, ...
- (missed)
- this is likely going to be a contentious fork. why should miners support the treasury chain instead of keeping the existing chain alive in which they are not taxed?
  - charles: this chain has a long track record without many activity; users are incentivized to move to the new chain
- alex: what about an opt-out flag to burn treasury rewards
  - charles: interesting, this is giving a vote to the miner, we can add this to the ecip
- (missed)
- afri proposes slido q&a event code #etctreas to priotize questions: https://app.sli.do/event/fxvuq4xl
- charles: mantis will be ready by november (???)
- charles: will write ecip
- terry: can charles give an honest assessment of why the treasury proposal was rejected last time it was proposed?
  - charles: people wanted to run the experiment decentralized funding, same as bitcoin
  - charles: "are you happy? or do you want a change, i'm proposing a change?"
- how is etc treasury proposal different than decred's current treasury system? it sounds very similar.
  - charles: try something different
- how does adding another client (mantis) solve any of our current issues, does it not introduce issues like we just faced with the chainsplit?
  - solving 51% is pretty straight forward, checkpointing!
  - innovate on the proof-of-work side (?)
  - don't develop a monoculture
- voting would be weighted by your stake, correct? we have some accounts that have a lot of etc and could hence themselves be the supermajority, no?
  - yes, could do stake-weighted voting
  - beacuse it's pow, it could include a voice of the miners
  - also could do reputation metrics
  - charles is not in favor of plutacratic systems

ecip discussion:
- yaz asked who is pro treasury, any opinion?
- alex thinks it's a great idea now
- ecc & etc core have contributed to the etc network for the past several years with real fund, while iohk left for a long time. why should iohk get fund (3m) from miners, not others? why should we trust iohk? ... (interrupted)
- charles: we have been here from the beginning
- terry: let him finish the question
- yaz: (continues) ...  can charles contribute to etc network with his personal assets for 1-2 years starting right now?
- charles: we have spent millions and brand in etc... is this even a legitimate question?
- charles: we need to innovate, this is our only option available to us
- alex: etc labs is rude and opaque in funding
- terry: why would we take away from the miners?
- charles: well, everyone is taxed. we tried no-funding approach for all the years. it didn't work. we don't have unlimited time.
- yaz: 6 minute over time
- yaz: why should we trust etc labs?
- terry: it's not about trust, we are going to contribute to etc regardless.
- zane: what should we look at for an accurate take on what this treasury proposal is? you mentioned that the 2017 white paper is stale. if so what's changed?
- charles: we'll write an ecip. hopes that on-chain changes will be mild. the actual complexity lives in a solidity smart contract.

yaz closes the call.
- yaz: next week we'll have a call about how to mitigate 51% attacks
- bob: wanted to express support, for the simple reason that growing etc is the root solution for a lot of problems
- alex: (missed)
- stev: (missed)
- hudson: has the zcash style major grants committee structure been explored at all by those proposing the treasury?
- charles: (sorry charles, you use so many words to describe these things)
(i'm stopping to write protocol after 82 minutes)

appendix - unanswered chat questions:
- so you see the future of etc as pow/pos hybrid, similar to decred? (no answer)
- who will own the pkeys of the smart contract?? how will a etclabs github take over be prevented??? (no answer)
- what about funding for the developer outside of those 3-4 teams? (no answer)
- when will etc network be safe so that exchanges can accept transactions? right now etc is under attack by everyone and there is a chance that exchanges will delist etc! we need to act very fast. (no answer)
- doesn't charles have a conflict of interest? - he is a cardano founder, iohk ceo, but yet he is helping another project that most would consider a competitor to cardano? (no answer)
- dapps devs getting funded was mentioned. however the treasury onlyu funds 3 entities. how are dapp devs assumed to receive funds?
would iohk accept a different treasury proposal - where three/four teams are not funded off the bat but instead a fund is built and paid to teams based on projects/proposals
- why is my question not asked? (no answer)
- how can we hold treasury system participants accountable if they miss deadlines or do not produce results beneficial to chain security? (no answer)
- how does adding another client (mantis) solve any of our current issues, does it not introduce issues like we just faced with the chainsplit? (no answer)
- thoughts on rebranding the etc name to something different?  especially all the negative press, and the changes that will happen to etc. (no answer)
- what is the purpose of a treasury? also, heck i'm in time out? what? (no answer)
- if i release some software tonight, that makes running nodes with existing functional clients, easier (~2 clicks for a core-geth node). can i replace iohk in the intial treasury, until they deliver a working mantis client? (no answer)
- what happens if the dev reward is too large due to etc price appreciation? (no answer)
- how to solve the conflict of interests of ada & etc? can charles step down at cardano and focus on etc? (no answer)
- why cant a treasury be done layer2 like the #communityfund, we have funding in the ecosystem, even myself would donate to a treasury on layer2 if i knew it was actually going to be used instead of sit there (no answer)
- what about the other dev teams that left? are they guarenteed x/x if they return and make clients? (no answer)
- no ones explained how the funding reaches dapp devs. can anyone elaborate on this? (no answer)
- if igor returns with etcdev and emerald - should he get funding? (no answer)
- what if etc price goes to the moon, do you need that much money? (no answer)
- what affect did the 51% have on the price? what effect would a treasury have on the price? (no answer)

more questions: https://app.sli.do/event/fxvuq4xl/live/questions

**Comment by TheEnthusiasticAs** [comment]
The audio record of the call provided by @developerkevin from ETC Cooperative https://www.youtube.com/watch?v=8EOz79tU_xo

**Comment by jackispm** [comment]
First of all, I am in favor of Treasury, but I don't agree with this preliminary draft( put forwards byt charles )

1.Treasury is used to fund future teams. Don't tell ETC what you did in the past or when you started to support ETC, it is not the reason why you should get funding. 

2. Treasury should not provide equal funding for several specific teams. Each team has different contributions, there are also high and low levels of competence among developers, why should Treasury divide the funding equally? That doesn't make sense. Team A works really hard, and team B just writes A few docs and copies A few lines of code, why do they have the same fund?

2.It's ok to build Treasury with smart contracts, but don't expect it to be able to evaluate the contributions of different teams.

 "Talk is cheap, show me the code".

5.Why are these teams getting funding? Who are these teams (A legitimate enterprise or an unregistered organization)? How can we trust their financial disclosure? How many developers do they have? Do they have a public resume? Why can't funding support individual developers or non-development teams? Why doesn't the fund go directly to commercial enterprises to pay for development?

6.Charles used tax to compare Treasury, thinking that it was miners' expense for security. I think it's ridiculous. The Internet promises safety first,  then miners participate in it, if it is a tax for security, the Treasury should tax exchanges and users, not miner.

7.If Treasury is designed to motivate developers, are non-developers in ETC lab or etc-cope eligible for funding? For example, secretaries, clerks, accountants, bloggers. If they are eligible for funding, and the ETC community has a large number of individual contributors, why can't they get funding?



To sum up, I believe that to achieve a fair Treasury system, there must be committees and voting rights, and mining pools and exchanges should be involved to prevent fraud in the name of development.

DON’T DON’T DON’T 
Don't try to use Treasury as an opportunity to cash out, no matter what you've paid in the past.

**Comment by q9f** [comment]
Closing till we have a proposal.
